machine/v1beta1/types_awsprovider.go

```diff
@@ -393,3 +399,46 @@ const (
 	// When set to CapacityBlock the instance utilizes pre-purchased compute capacity (capacity blocks) with AWS Capacity Reservations.
 	MarketTypeCapacityBlock MarketType = "CapacityBlock"
 )
+
+type HostPlacement struct {
+	// affinity specifies the affinity setting for the instance.
+	// When Affinity is set to DedicatedHost, an instance started onto a specific host always restarts on the same host if stopped. This setting also requires the configuration of DedicatedHost.
+	// When Affinity is set to AnyAvailable, and you stop and restart the instance, it can be restarted on any available host.
+	// When Affinity is set to DedicatedHost, dedicated field is required.
+	// The default value is AnyAvailable
+	// +kubebuilder:default=AnyAvailable
+	// +openshift:enable:FeatureGate=AWSDedicatedHosts
+	// +optional
+	Affinity *HostAffinity `json:"affinity,omitempty"`
+
+	// dedicated specifies a particular dedicated host when required by affinity set to DedicatedHost.
+	// Must be omitted when hostAffinity is "AnyAvailable".
```

Comment:
We prefer to use the terminology:
> dedicated is required when 'affinity' is set to `DedicatedHost`, and forbidden otherwise. 

Additionally, we should add CEL validation that enforces this behavior.
{TODO: Link to example}

Reply:
I'll update the parent to have this information in addition to the tags mentioned in the example for the discriminator and members.